experimental/web/sample_webgpu/main.c

```diff
@@ -66,9 +69,11 @@ const char* call_function(iree_program_state_t* program_state,
 typedef struct iree_sample_state_t {
   iree_runtime_instance_t* instance;
   iree_hal_device_t* device;
+  iree_hal_allocator_t* heap_allocator;
```

Comment:
expose iree_hal_heap_buffer_wrap in buffer.h - we have the subspan buffer already exposed so it's a natural fit: https://github.com/google/iree/blob/main/runtime/src/iree/hal/buffer.h#L723-L726
the only thing this allocator does is call that (and mess up the params, which we really don't want)
that'll also let you provide a release callback that you can put the wgpuBufferUnmap call in, ensuring that the wrapped memory remains valid for the lifetime of the buffer view